pkg/controller/utils/elasticsearch.go

```diff
@@ -62,3 +102,323 @@ func GetElasticsearchClusterConfig(ctx context.Context, cli client.Client) (*ren
 
 	return render.NewElasticsearchClusterConfigFromConfigMap(configMap)
 }
+
+type ElasticClient interface {
+	SetILMPolicies(*operatorv1.LogStorage) error
+	GetClient() *elastic.Client
+}
+
+type EsClient struct {
+	Ctx    context.Context
+	Client *elastic.Client
+}
+
+// GetElasticClient creates new Elastic client and returns ElasticClient interface
+func GetElasticClient(client client.Client, ctx context.Context) (ElasticClient, error) {
+	esClnt, err := NewClient(client, ctx)
+	if err != nil {
+		return nil, err
+	}
+	es := EsClient{ctx, esClnt}
+	return &es, nil
+}
+
+// SetILMPolicies creates ILM policies for each timeseries based index using the retention period and storage size in LogStorage
+func (es *EsClient) SetILMPolicies(ls *operatorv1.LogStorage) error {
+	policyList := es.ListILMPolicies(ls)
+	return es.CreateOrUpdatePolicies(policyList)
+}
+
+func (es *EsClient) ListILMPolicies(ls *operatorv1.LogStorage) map[string]PolicyDetail {
```

Comment:
I think it can be rewritten to
```
func (es *EsClient) ListILMPolicies(ls *operatorv1.LogStorage) map[string]PolicyDetail {
	totalEsStorage := getTotalEsDisk(ls)
	pctOfTotalDisk := 0.7
	pctOfDiskAuditSnapshot := 0.1 / float64(6)
	
	return map[string]PolicyDetail{
		"tigera_secure_ee_flows" : BuildIlmPolicy(totalEsStorage, pctOfTotalDisk, 0.9, int(*ls.Spec.Retention.Flows)),
		"tigera_secure_ee_dns" : BuildIlmPolicy(totalEsStorage, pctOfTotalDisk, 0.05, 8),
		"tigera_secure_ee_bgp" : BuildIlmPolicy(totalEsStorage, pctOfTotalDisk, 0.05, 8),

		"tigera_secure_ee_audit_ee" : BuildIlmPolicy(totalEsStorage, pctOfTotalDisk, pctOfDiskAuditSnapshot, int(*ls.Spec.Retention.AuditReports)),
		"tigera_secure_ee_audit_kube" : BuildIlmPolicy(totalEsStorage, pctOfTotalDisk, pctOfDiskAuditSnapshot, int(*ls.Spec.Retention.AuditReports)),
		"tigera_secure_ee_snapshots" : BuildIlmPolicy(totalEsStorage, pctOfTotalDisk, pctOfDiskAuditSnapshot, int(*ls.Spec.Retention.Snapshots)),
		"tigera_secure_ee_benchmark_results" : BuildIlmPolicy(totalEsStorage, pctOfTotalDisk, pctOfDiskAuditSnapshot, 91),
		"tigera_secure_ee_events" : BuildIlmPolicy(totalEsStorage, pctOfTotalDisk, pctOfDiskAuditSnapshot, 91),
		
	}
}
```
And then policiesForIndicesWithMajorSpace and policiesForIndicesWithMinorSpace can be removed.